### PR TITLE
Ottano SITL

### PR DESCRIPTION
**Pull request by robertlong13**

### CODE REVIEW

**Comment by robertlong13** [comment]
Marking this ready for review. @Pradeep-Carbonix @loki077 

Of note, I have made a change to CX_BIT in here. The change is non-functional in practice, but it is the only change in here that impacts any bytes in our firmware builds (everything else is either purely in SITL areas of cpp, or blocked off beind #ifs).

I will now work on the CX_BIT autotest, which I will base on top of this PR.

It's a large PR. I recommend filtering to view one or two commits at a time. Relevant upstreams that I have opened during this too:
https://github.com/ArduPilot/ardupilot/pull/28904
https://github.com/ArduPilot/ardupilot/pull/28895
https://github.com/ArduPilot/ardupilot/pull/28905

File: Tools/Carbonix_scripts/process_sitl_defaults.py [hunk 1]
```diff
@@ -39,11 +39,23 @@ def process_defaults(file, depth=0):
             continue
 
         if line.startswith("@delete"):
-            pattern = line.split(maxsplit=1)[1]
-            for i in range(len(param_list)):
-                param_name = re.split(r"[\s,]+", param_list[i])[0]
+            # Strip trailing comments
+            line = re.sub(r"\s*#.*$", "", line)
+            line.strip()
```
Comment (loki077) [inline Tools/Carbonix_scripts/process_sitl_defaults.py]: shouldn't it be line = line.strip()

Comment (robertlong13) [reply Tools/Carbonix_scripts/process_sitl_defaults.py]: Yes, good catch

File: Tools/Carbonix_scripts/sitl_engine_curve_fit.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+# CHT and CHT2 Fit Parameters
+# These parameters define the thermal behavior of the engine:
+# - FULL_DT_CRUISE: Full throttle delta temperature at cruise speed (°C).
+# - IDLE_DT_CRUISE: Idle throttle delta temperature at cruise speed (°C).
+# - IDLE_DT_HOVER: Idle throttle delta temperature while hovering (°C).
+# - TCONST_CRUISE: Time constant for thermal inertia at cruise speed (s).
+#   Higher values mean more thermal inertia.
+# - KAPPA: Scaling factor for how thermal inertia changes with airspeed.
+#   Increading this will make on-ground warmup take longer.
```
Comment (loki077) [inline Tools/Carbonix_scripts/sitl_engine_curve_fit.py:67]: typo Increasing.

Comment (loki077) [inline Tools/Carbonix_scripts/sitl_engine_curve_fit.py]: typo Increasing.

File: Tools/Carbonix_scripts/sitl_engine_curve_fit.py [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+simulations. It handles calibration for cylinder head temperature (CHT and
+CHT2) and fuel consumption using flight log data. The calibration process
+optimizes key fit parameters to minimize the difference between actual and
+simulated values from the log data.
+
+Features:
+  - Independent calibration for CHT, CHT2, and fuel consumption.
+  - Visualization of actual vs simulated results for all three metrics.
+
+Input CSV File Format Requirements: The input file should be a CSV file with
```
Comment (loki077) [inline Tools/Carbonix_scripts/sitl_engine_curve_fit.py:14]: any particular reason you not directly processing a binary log? 

Also, putting comment on how to extract this info from binary would be handy.

Comment (robertlong13) [reply Tools/Carbonix_scripts/sitl_engine_curve_fit.py:14]: I generate the csv with UAVLogViewer. I plot all the variables of interest, and then hit the "export data" button.

Three big reasons:
1. Much easier to get resampled data this way, with all columns sampled at the same evenly-spaced time interval
2. Easier to select the region of interest that I want to fit.
3. Way less data to import when I was running repeatedly during dev.

Comment (loki077) [reply Tools/Carbonix_scripts/sitl_engine_curve_fit.py:14]: That make sense. Also didn't realised we can export from UAVlogViewer.

File: Tools/Carbonix_scripts/process_sitl_defaults.py (hunk 1 above)
Comment (robertlong13) [reply Tools/Carbonix_scripts/process_sitl_defaults.py]: This line was actually unnecessary, which is why I didn't catch that it wasn't doing what it was supposed to. The `split` already harmlessly removes trailing split characters.

**Review by loki077 — APPROVED** [review]
LGTM! Great job on this PR, @robertlong13. I really liked ESC and EFI simulation implementation makes things feel a lot more realistic.

I’ve suggested a few minor changes. Haven’t had a chance to do a full run all the sim, but I’ll test it tomorrow in the office. Overall, the code flow looks solid!

**Pull request merged**